### ISSUE TITLE
Problem using Azure principal account to deploy workbench application through workbench rest apis

### ISSUE THREAD

**Issue by ygizhitsa** [Azure-Samples/blockchain#101]
Using Azure principal account to deploy workbench application using workbench rest apis

## Minimal steps to reproduce
>
1. Create Azure Principal account (az ad sp create for rbac)
2. In Azure DevOps ARM service connection with principal credentials
3. Create Azure DevOps pipeline 
4. In release phase create rest calls to the Azure Workbench api to deploy sol and json artifacts 

### Any log messages given by the failure
>

### Expected/desired behavior
>
Principal account would be able to register new application

### OS and Version?
>Azure DevOps -> Azure Blockchain Workbench

### Versions
>1.5

### Mention any other details that might be useful

> Could not aff principal to a workbench administrators groups as a result could not register new Workbench application 
  
> Thanks! We'll be in touch soon.

**Comment by ygizhitsa** [comment]
Ali, I have used  az ad sp  for rbac to create principal along with registered application. Let me know if I could assist somehow

**Comment by ali92hm** [comment]
Hey @ygizhitsa, sorry for getting back to you late. There has been a lot of asks and confusion about this so I took some time to write a document to clarify.

Usually when you run the command `az ad sp` that service principal will be created in your native active directory and since Workbench typically workbench uses another active directory, your SP won't have access to Workbench.

In addition, Admin level access is required for the SP to create an application in Workbench. Assigning a SP as an Admin user is not straight forward.

I've created this[ script and document ](https://github.com/Azure-Samples/blockchain/tree/master/blockchain-workbench/scripts/workbench-serviceprincipal)to solve this problem, give it a try and let me know if there is anything missing.
